client/packages/programs/src/JsonForms/components/PeriodSearch.tsx

```diff
@@ -29,11 +32,23 @@ const UIComponent = (props: ControlProps) => {
   const programId = options?.findByProgram
     ? extractProperty(core?.data, 'programId')
     : null;
-  const { data, isLoading } = usePeriodList(
+  const today = new Date();
+  const lastDayOfMonth = DateUtils.lastDayOfMonth(today);
+  const { data, isFetching, fetchNextPage } = usePeriodList(
+    RECORDS_PER_PAGE,
     programId,
-    options?.findByProgram ? !!programId : true
+    options?.findByProgram ? !!programId : true,
+    {
+      endDate: {
+        beforeOrEqualTo: Formatter.naiveDate(lastDayOfMonth),
```

Comment:
Want to get all periods until the last day of current month in case there are transactions done in this time

Reply:
This is great but I think it misses some edge cases:

If a period is monthly but doesn't begin on the first day of the month, then it won't capture this period (in my case I had a monthly period schedule which began on the 10th, and wasn't visible). I'm not sure if this is a common case, but there it is.

I wonder if it would be better to filter instead by startDate and filter for startDate < now.

I recognise requires refactoring of backend filters, but makes for a more simple front end use.

Reply:
Yup good idea